sktime/classifiers/proximity.py

```diff
@@ -0,0 +1,1084 @@
+    ----
+    result : boolean
+        whether the set of class labels is pure
+    '''
+    # get unique class labels
+    unique_class_labels = np.unique(class_labels)
+    # if more than 1 unique then not pure
+    return len(unique_class_labels) <= 1
+
+
+def gini(parent_class_labels, children_class_labels):
+    '''
+    get gini score of a split, i.e. the gain from parent to children
+    ----
+    Parameters
+    ----
+    parent_class_labels : 1d numpy array
+        array of class labels at parent
+    children_class_labels : list of 1d numpy array
+        list of array of class labels, one array per child
+    ----
+    Returns
+    ----
+    score : float
+        gini score of the split from parent class labels to children. Note the gini score is scaled to be between 0
+        and 1. 1 == pure, 0 == not pure
+    '''
+    # find gini for parent node
+    parent_score = gini_node(parent_class_labels)
+    # find number of instances overall
+    parent_num_instances = parent_class_labels.shape[0]
+    # sum the children's gini scores
+    children_score_sum = 0
+    for index in range(0, len(children_class_labels)):
+        child_class_labels = children_class_labels[index]
+        # find gini score for this child
+        child_score = gini_node(child_class_labels)
+        # weight score by proportion of instances at child compared to parent
+        child_size = len(child_class_labels)
+        child_score *= (child_size / parent_num_instances)
+        # add to cumulative sum
+        children_score_sum += child_score
+    # gini outputs relative improvement
+    score = parent_score - children_score_sum
+    return score
+
+
+def gini_node(class_labels):
+    '''
+    get gini score at a specific node
+    ----
+    Parameters
+    ----
+    class_labels : 1d numpy array
+        array of class labels
+    ----
+    Returns
+    ----
+    score : float
+        gini score for the set of class labels (i.e. how pure they are). 1 == pure, 0 == not pure
+    '''
+    # get number instances at node
+    num_instances = class_labels.shape[0]
+    score = 1
+    if num_instances > 0:
+        # count each class
+        unique_class_labels, class_counts = np.unique(class_labels, return_counts = True)
+        # subtract class entropy from current score for each class
+        for index in range(0, len(unique_class_labels)):
+            class_count = class_counts[index]
+            proportion = class_count / num_instances
+            sq_proportion = np.math.pow(proportion, 2)
+            score -= sq_proportion
+    # double score as gini is between 0 and 0.5, we need 0 and 1
+    score *= 2
+    return score
+
+
+# todo info gain
+def information_gain(parent_class_labels, children_class_labels):
+    raise Exception('not implemented yet')
+
+
+# todo chi sq
+def chi_squared(parent_class_labels, children_class_labels):
+    raise Exception('not implemented yet')
+
+
+def pick_one_exemplar_per_class(instances, class_labels, rand):
```

Comment:
For handling the random number generation here and throughout, I would rely on sklearn's [`check_random_state()`](http://scikit-learn.org/stable/modules/generated/sklearn.utils.check_random_state.html)

Reply:
Ah great method, thanks for the pointer! :)